frontend/src/components/session-header/session-header.tsx

```diff
@@ -28,16 +28,41 @@ const SessionHeader = (): ReactElement => {
 
   return (
     <div className="session-header">
-      <Link to="/">
+      {state.sessionInProgress ? (
         <img
           alt="UCF React Logo"
           src="/img/UCFReactLogoBlackBackground.png"
-          onClick={clearSession}
+          onClick={() => {
+            dispatch({ type: "show-exit-warning-modal" });
+            clearSession();
```

Comment:
This is an issue. you shouldn't be opening the modal and clearing the session. you should be clearing the session if they hit confirm on the modal. Meaning you probably have to copy the `clearSession` functionality to the modal component and call it there if they hit confirm